core/chatListener.go

```diff
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"errors"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gabek/owncast/core/chat"
+	"github.com/gabek/owncast/models"
+)
+
+//ChatListenerImpl the implementation of the chat client
+type ChatListenerImpl struct{}
```

Comment:
Yeah, it's kind of lousy we have to add another interface + implementation in order to work around that, but I don't have a better solution offhand.  It leaves it better than it was, and if we come up with something better we can change it.  I'm actually assuming the chat feature will get refactored in the future to have/require more functionality and become more complex, but we can cross that bridge when we come to it.

Reply:
Yeah, that's what I am thinking. This will work for now and we can absolutely change it to something 100x better when refactored.